### PR TITLE
Fixing Copy Mode

### PR DESCRIPTION
**Pull request by eyal-sasson**
Switched from using `select` to using decorations in xterm.js.

### CODE REVIEW

**Comment by daonb** [comment]
The pane looks good, but it doesn't copy the selection to the clipboard

File: vite.config.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ export default defineConfig({
       reporters: "verbose",
       globals: true,
       exclude: [ ...configDefaults.exclude, "aatp/*/*" ],
-      setupFiles: "vitest.setup.ts"
+      // setupFiles: "vitest.setup.ts"
```
Comment (daonb) [inline vite.config.js:14]: sorry about that, I've just pushed the file

File: src/pane.js [hunk 2]
```diff
@@ -865,32 +871,80 @@ export class Pane extends Cell {
         }
     }
     cmInitCursor() {
-        var selection = this.t.getSelectionPosition()
+        /* var selection = this.t.getSelectionPosition()
```
Comment (daonb) [inline src/pane.js:874]: Please remove this and the next 7 lines

**Review by daonb — CHANGES_REQUESTED** [review]
I found two issues: It doesn't copy the selection to the clipboard and when scrolling the page I lose the cursor.

File: src/pane.js [hunk 3]
```diff
@@ -586,14 +592,27 @@ export class Pane extends Cell {
...
+        if (lines.length == 0)
+            return
+        if (lines.length == 1)
+            lines[0] = lines[0].slice(this.cmSelection.startColumn, this.cmSelection.endColumn + 1)
+        else {
+            lines[0] = lines[0].slice(this.cmSelection.startColumn)
+            lines[lines.length - 1] = lines[lines.length - 1].slice(0, this.cmSelection.endColumn + 1)
+        }
+        console.log(lines.join('\n'))
+        return Clipboard.write({string: lines.join('\n')})
```
Comment (daonb) [inline src/pane.js:611]: It still copy garbage to the clipboard like `mw[3` . Take a look at how we handle pointer up and copy the selection to the Clipboard

**Review by daonb — CHANGES_REQUESTED** [review]
It can scroll now, but doesn't copy valid data to the clipboard

File: src/pane.js (hunk 3 above)
Comment (eyal-sasson) [reply src/pane.js:611]: oh we do have `buffer.activeg.getLine`, oops

Comment (daonb) [reply src/pane.js:611]: When a library call like `Clipboard.write` doesn't work as expected I always start by searching T7 code base. Chances are we use it in other place.

Comment (eyal-sasson) [reply src/pane.js:603]: we were using it before, hopefully it works now

**Pull request merged**